src/psyclone/tests/domain/lfric/lfric_scalar_test.py

```diff
@@ -105,25 +455,474 @@ def test_int_real_scalar_invalid():
             "This is invalid." in str(err.value))
 
 
-def test_scalar_invoke_uniq_declns_valid_intrinsic():
-    ''' Tests that all valid intrinsic types for user-defined scalar arguments
-    ('real' and 'integer') are accepted by Invoke.unique_declarations().
+# Tests for PSy-layer code generation for scalar arguments
```

Comment:
Again, I think we should keep the meta-data tests separate from the code-generation tests.

Reply:
Split the scalar tests into `lfric_scalar_mdata_test.py` and `lfric_scalar_codegen_test.py`.